projects/ccd-case-ui-toolkit/src/lib/shared/services/fields/fields.purger.ts

```diff
@@ -309,10 +309,16 @@ export class FieldsPurger {
    * @returns A new array with the mapped values
    */
   public mapArrayValuesToNull(array: any[]): any[] {
-    return array.map(element => {
-      return typeof element === 'object'
-        ? Object.assign({}, ...Object.keys(element).map(k => ({ [k]: null })))
-        : null;
-    });
+    if (array && array.length > 0) { FieldsUtils.isArray([])
+      return array.map(element => {
+        if (element !==  undefined  && element !== null) {
+          return typeof element === 'object'
+            ? Object.assign({}, ...Object.keys(element).map(k => ({ [k]: null })))
+            : null;
+        }
```

Comment:
@OgunyemiO are you not checking element if it is null or undefined?

Reply:
@RiteshHMCTS  The check is there:

 if (element !==  undefined  && element !== null) {
          return typeof element === 'object'
            ? Object.assign({}, ...Object.keys(element).map(k => ({ [k]: null })))
            : null;
        }

Reply:
@OgunyemiO there is if condition for element is not undefined or not null but no else condition